extensions/ipynb/src/ipynbMain.ts

```diff
@@ -6,6 +6,25 @@
 import * as vscode from 'vscode';
 import { NotebookSerializer } from './notebookSerializer';
 
+// From {nbformat.INotebookMetadata} in @jupyterlab/coreutils
+type NotebookMetadata = {
+	kernelspec?: {
+		name: string;
+		display_name: string;
+		[propName: string]: unknown;
+	};
+	language_info?: {
+		name: string;
+		codemirror_mode?: string | {};
+		file_extension?: string;
+		mimetype?: string;
+		pygments_lexer?: string;
+		[propName: string]: unknown;
+	};
+	orig_nbformat: number;
```

Comment:
Do we want to expose that as part of the API? In my mind the notebook serial format is "owned" by the ipynb extension, so I would expect that only it is allowed to manipulate this field?

Reply:
I'm leaving this as this API is about updating the notebook metadata.
Personally see no harm in leaving this, don't want to add code to explicitly disallow this item.